bin/mariadb

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use lib '/usr/local/pf/lib';
+
+use pf::services::util;
+use POSIX qw(:signal_h pause _exit setuid setgid);
+use pf::log;
+use pf::cluster;
+use pf::config qw(
+    %Config
+);
+use pf::util;
+use DBI;
+
+our $PROGRAM_NAME = $0 = "pf-mariadb";
+
+my $logger = get_logger( $PROGRAM_NAME );
+
+our $RUNNING = 1;
+
+my (undef, undef,$uid,$gid) = getpwnam('mysql');
+setuid($uid);
+setgid($gid);
+
+our $MYSQLD_PID;
+our $IS_CHILD;
+
+# How many times should we attempt to connect to a galera cluster when an alive quorum is there
+# After this, the management node will declare a new cluster if its not in maintenance mode
+# TODO: move this to configuration
+our $PEERS_ALIVE_CONNECT_ATTEMPT = 1;
+
+# Time to wait between starts after a failure
+# TODO: move this to configuration
+our $WAIT_TIME_FAIL = 10;
+
+# Time to wait between attempts to discover alive servers
+# TODO: move this to configuration
+our $WAIT_TIME_ALIVE = 10;
+
+our $failed_counts = {};
+
+# init signal handlers
+POSIX::sigaction(
+    &POSIX::SIGHUP,
+    POSIX::SigAction->new(
+        'normal_sighandler', POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    )
+) or $logger->logdie("pf-mariadb: could not set SIGHUP handler: $!");
+
+POSIX::sigaction(
+    &POSIX::SIGTERM,
+    POSIX::SigAction->new(
+        'normal_sighandler', POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    )
+) or $logger->logdie("pf-mariadb: could not set SIGTERM handler: $!");
+
+POSIX::sigaction(
+    &POSIX::SIGINT,
+    POSIX::SigAction->new(
+        'normal_sighandler', POSIX::SigSet->new(), &POSIX::SA_NODEFER
+    )
+) or $logger->logdie("pf-mariadb: could not set SIGINT handler: $!");
+
+my $STARTED_BECAUSE;
+
+=head2 record_started_because
+
+Increment a counter for a starting strategy
+
+=cut
+
+sub record_started_because {
+    my ($started_because) = @_;
+    if(!$started_because) {
+        die "Missing starting reason";
+    }
+
+    $STARTED_BECAUSE = $started_because;
+
+}
+
+=head2 launch_mysql
+
+Launch MySQL/MariaDB with a specific set of arguments
+
+=cut
+
+sub launch_mysql {
+    my (@args) = @_;
+    
+    my $args = join(' ', @ARGV, @args);
+    # Must not lookup in /sbin for mysqld, but in /usr/sbin
+    $ENV{PATH} = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin:/root/bin";
+    my $cmd = "mysqld_safe --defaults-file=/usr/local/pf/var/conf/mariadb.conf $args";
+    print "Starting MySQL with command: $cmd \n";
+    `$cmd`;
```

Comment:
Are we sure this does not call /bin/sh in the background?
Meaning we have an additional process between us?